ui/js/controllers/perf/compare.js

```diff
@@ -27,6 +23,20 @@ perf.controller('CompareChooserCtrl', [
                                        $stateParams.originalRevision : '');
             $scope.newRevision = ($stateParams.newRevision ?
                                   $stateParams.newRevision : '');
+            if(localStorageService.get('originalProject') !== null){
+                $scope.originalProject = localStorageService.get('originalProject');
+                console.log("Local Storage");
+            }
+            else{
+                $scope.originalProject = _.findWhere(projects, {
+                    name: ($stateParams.originalProject ?
+                        $stateParams.originalProject : phCompareDefaultOriginalRepo)
+                }) || projects[0];
```

Comment:
If you just use localStorage to store the name of the original/new projects (as suggested below), you can use this type of logic to look up the right object. We can just vary whether to use what's in local storage vs. phCompareDefaultOriginalRepo depending on whether the former is defined.

Does that make sense?